### PR TITLE
Add user mode changes in channels

### PR DESCRIPTION
**Pull request by CoryChaplin**
TODO : manage channel modes

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Hey @CoryChaplin, is the TODO for this PR or a future one?

File: src/channel.js [hunk 1]
```diff
@@ -61,6 +61,36 @@ function IrcChannel(irc_client, channel_name, key) {
...
+        if (event.target === that.name) {
+            // There can be multiple modes set at once, loop through
+            _.each(event.modes, function(modes) {
+                // Is it a channel mode ?
+                if(modes.param === null) {
+                    // TODO : manage channel mode changes
+                } else { // It's a user mode
+                    // Find the user affected
+                    _.find(that.users, function(user) {
+                        if(user.nick.toLowerCase() === modes.param.toLowerCase()) {
```
Comment (MaxLeiter) [inline src/channel.js]: `if(` -> `if (`

File: src/channel.js [hunk 2]
```diff
@@ -61,6 +61,36 @@ function IrcChannel(irc_client, channel_name, key) {
             }
         });
     });
+    irc_client.on('mode', function(event) {
+        if (event.target === that.name) {
+            // There can be multiple modes set at once, loop through
+            _.each(event.modes, function(modes) {
+                // Is it a channel mode ?
+                if(modes.param === null) {
```
Comment (MaxLeiter) [inline src/channel.js]: `if(` -> `if (`

File: src/channel.js [hunk 3]
```diff
@@ -61,6 +61,36 @@ function IrcChannel(irc_client, channel_name, key) {
...
+                    // TODO : manage channel mode changes
+                } else { // It's a user mode
+                    // Find the user affected
+                    _.find(that.users, function(user) {
+                        if(user.nick.toLowerCase() === modes.param.toLowerCase()) {
+                            _.each(modes, function(mode) {
+                                if(mode.substring(0,1) === '+') {
```
Comment (MaxLeiter) [inline src/channel.js]: `if(` -> `if (` for rest of these

**Comment by CoryChaplin** [comment]
The TODO is for later.

**Comment by CoryChaplin** [comment]
@MaxLeiter Fixed

**Comment by MaxLeiter** [comment]
@CoryChaplin let me and or @prawnsalad know when this is good to go

**Pull request merged**